aiohomekit/controller/ble/pairing.py

```diff
@@ -276,6 +276,8 @@ async def _ensure_connected(self):
         if self.client and self.client.is_connected:
             return
         async with self._connection_lock:
+            if self._shutdown:
+                raise RuntimeError("Pairing is shutting down")
```

Comment:
Will change this to return instead since the race does happen because the first check happens and then we await and can still get here after shutdown

```
2022-09-14 10:01:41.511 ERROR (MainThread) [aiohomekit.utils] Failure running background task: Task-2266
Traceback (most recent call last):
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/utils.py", line 29, in _handle_task_result
    task.result()
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/pairing.py", line 382, in _process_disconnected_events
    results = await self.get_characteristics(list(self.subscriptions))
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/client.py", line 75, in _async_wrap
    return await func(*args, **kwargs)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/pairing.py", line 676, in get_characteristics
    return await self._get_characteristics_without_retry(characteristics)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/pairing.py", line 109, in _async_wrap
    return await func(self, *args, **kwargs)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/pairing.py", line 683, in _get_characteristics_without_retry
    await self._populate_accessories_and_characteristics()
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/pairing.py", line 552, in _populate_accessories_and_characteristics
    await self._ensure_connected()
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/pairing.py", line 280, in _ensure_connected
    raise RuntimeError("Pairing is shutting down")
RuntimeError: Pairing is shutting down
```